### DISCUSSION TITLE
Better thumbnail control

### DISCUSSION THREAD

**Discussion by myCodebox** [pocketbase/pocketbase#162] [Q&A] [answered]
It would be great if you could also specify such data for a thumbnail.

```
w100 # aspect ratio [height auto / width 100]
h100 # aspect ratio [height 100 / width auto]
```

You can easily add that with `github.com/disintegration/imaging`.
```golang
// Resize srcImage to width = 800px preserving the aspect ratio.
dstImage800 := imaging.Resize(srcImage, 100, 0, imaging.Lanczos)
```

I tried to change it locally:
https://github.com/pocketbase/pocketbase/blob/master/tools/filesystem/filesystem.go#L208
```golang
var width, height int
var resize bool = false

if strings.HasPrefix(thumbSize, "w") {
width, _ = strconv.Atoi(strings.Replace(thumbSize, "w", "", -1))
resize = true
}

if strings.HasPrefix(thumbSize, "h") {
height, _ = strconv.Atoi(strings.Replace(thumbSize, "h", "", -1))
resize = true
}

thumbSizeParts := strings.SplitN(thumbSize, "x", 2)
if len(thumbSizeParts) == 2 {
width, _ = strconv.Atoi(thumbSizeParts[0])
height, _ = strconv.Atoi(thumbSizeParts[1])
}
```
And at the image fill func, I expanded it like this
```golang
var thumbImg image.Image
if resize == true {
// create resize thumb imaging object
thumbImg = imaging.Resize(img, width, height, imaging.CatmullRom)
} else {
// create resize and crop thumb imaging object
thumbImg = imaging.Fill(img, width, height, cropAnchor, imaging.CatmullRom)
}
```
... and included in the validation of this one
https://github.com/pocketbase/pocketbase/blob/master/models/schema/schema_field.go#L461
```golang
validation.Field(&o.Thumbs, validation.Each(validation.Match(regexp.MustCompile(`^[1-9]\d*x[1-9]\d*|[w|h][1-9]\d*$`)))),
```
There were no errors during the build, even when creating the collection.

On call the thumb with `?thumb=w100` it sends the full size image.

Have I forgotten something?

**Comment by ganigeorgiev** [comment, marked as answer]
You need to specify the thumb in the related file field options.

In any case, there is already existing issue for expanding the thumbs manipulation api (see https://github.com/pocketbase/pocketbase/issues/87). Its in the roadmap (after the export/import feature) and we will need to decide on the thumb parameter format before proceeding (I still haven't review what other image manipulation apis are using).

**Comment by myCodebox** [comment]
Thanks for the info and sorry I missed the issue entry!
